### PR TITLE
FAT-960 mod-kb-ebsco-java: Implement API Karate tests - KB credentials

### PR DESCRIPTION
**Pull request by Serhii-Zdorenko**
**Purpose/Overview:**
Implement API Karate tests according to test plan

**Approach:**

https://wiki.folio.org/display/FOLIJET/Karate+test+plan+for+cucumber+report

**Acceptance** criteria:
Undefined scenarios added and shown in Cucumber report

### CODE REVIEW

File: mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature [hunk 1]
```diff
@@ -3,155 +3,566 @@ Feature: KB Credentials
...
+    And match responseType == 'json'
+    And match response.data[0].type == 'kbCredentials'
 
-  @Undefined
-  Scenario: GET /eholdings/user-kb-credential should return KB credentials by given assigned user (response: status 200 and json body)
-    * print 'undefined'
-
-  @Undefined
+  @Positive
   Scenario: POST /eholdings/kb-credentials should create KB credentials (response: status 201 and json body)
```
Comment (psmagin) [inline mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature:20]: You should check that the credentials entity was really created with data that you sent..

File: mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature [hunk 2]
```diff
@@ -3,155 +3,566 @@ Feature: KB Credentials
...
+    When method GET
+    Then status 200
+    And match response.attributes.name == 'NEW_NAME'
+
+    Given path '/eholdings/kb-credentials', id
+    When method DELETE
+    And status 204
+
+  @Positive
   Scenario: DELETE /eholdings/kb-credentials/{id} should should delete KB credentials by id (response: status 204)
```
Comment (psmagin) [inline mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature:98]: You should check that the credentials entity was really deleted.

File: mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature [hunk 3]
```diff
@@ -3,155 +3,580 @@ Feature: KB Credentials
...
+    And match responseType == 'json'
+
+    Given path '/eholdings/kb-credentials', id
+    When method DELETE
+    And status 204
+
+
+
+  @Negative
   Scenario: PATCH /eholdings/kb-credentials{id} should return status 400 and json if customerId invalid
```
Comment (oleksandr-dekin) [inline mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature:355]: should return status 400, but in the scenario I don't see response with 400

Comment (Serhii-Zdorenko) [reply mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature:355]: you are right, corrected

File: mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature [hunk 4]
```diff
@@ -3,155 +3,580 @@ Feature: KB Credentials
   Background:
     * url baseUrl
     * callonce login testUser
-    * configure headers = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': 'application/json, text/plain' }
+    * configure headers = { 'Content-Type': 'application/vnd.api+json', 'x-okapi-token': '#(okapitoken)', 'Accept': 'application/json, text/plain, application/vnd.api+json' }
+    * def credentials = read('classpath:domain/mod-kb-ebsco-java/features/samples/credentials/credentials.json')
+    * def random_uuid = 'dee3b52e-f37e-44f6-9d9c-681799c97543'
+    * def long_name = 'This name longer than 255 characters___This name longer than 255 characters___This name longer than 255 characters___This name longer than 255 characters___This name longer than 255 charactersThis name longer than 255 characters___This name longer than 255 characters___'
```
Comment (oleksandr-dekin) [inline mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature:9]:
```suggestion
    * def long_name = new Array(256).fill('a').join('')
```

**Review by oleksandr-dekin — APPROVED** [review]

**Review by Shans-Kaluhin — APPROVED** [review]
Good work!

**Pull request merged**